### DISCUSSION TITLE
SEQ instance with limited internet access using Azure Active Directory

### DISCUSSION THREAD

**Discussion by rykkart** [datalust/seq-tickets#1867] [Help]
Our SEQ is installed in a environment with limitted internet access. We are using Azure Active Directory as Authentication provider and need to know wich adresses you are using when trying do federate with Azure AD. Currently we are getting timeout during log in and getting the error "No extended description available" in SEQ UI.

**Comment by liammclennan** [comment]
Hi @rykkart 

Have a look at your Seq authentication configuration (Settings > System). The Azure AD authority is configurable there. 

If that doesn't help try looking through the Seq log (Settings > Diagnostics > Download a full diagnostic report). I expect the network failures will be logged there.

**Comment by rykkart** [comment]
Sadly not, the only thing I can see is this:

{"@t":"2023-06-12T07:33:42.2617790Z","@mt":"Exception occurred while processing message.","@l":"Error","@x":"System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.\r\n ---> System.IO.IOException: Unable to read data from the transport connection: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond..\r\n ---> System.Net.Sockets.SocketException (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.\r\n   --- End of inner exception stack trace ---\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\r\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)\r\n   at System.Net.Security.SslStream.<FillHandshakeBufferAsync>g__InternalFillHandshakeBufferAsync|182_0[TIOAdapter](TIOAdapter adap, ValueTask`1 task, Int32 minSize)\r\n   at System.Net.Security.SslStream.ReceiveBlobAsync[TIOAdapter](TIOAdapter adapter)\r\n   at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](TIOAdapter adapter, Boolean receiveFirst, Byte[] reAuthenticationData, Boolean isApm)\r\n   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)\r\n   --- End of inner exception stack trace ---\r\n   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore(Boolean async, Stream stream, SslClientAuthenticationOptions sslOptions, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)\r\n   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)\r\n   at System.Net.Http.HttpClient.SendAsyncCore(HttpRequestMessage request, HttpCompletionOption completionOption, Boolean async, Boolean emitTelemetryStartStop, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.RedeemAuthorizationCodeAsync(OpenIdConnectMessage tokenEndpointRequest)\r\n   at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleRemoteAuthenticateAsync()","EventId":{"Id":17,"Name":"ExceptionProcessingMessage"},"SourceContext":"Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler","RequestId":"8001ade0-0001-e800-b63f-84710c7967bb","RequestPath":"/aad"}

**Comment by liammclennan** [comment]
See if you can access your Azure AD authority (https://login.windows.net or similar) from the Seq server. If you can get that to work then you should be fine.

**Comment by jmezach** [comment]
We just ran into this issue as well. After checking the openid-configuration document we noticed that it contains references to https://graph.microsoft.com/ so we're guessing that Seq also needs to be able to access that domain.

**Reply by nblumhardt** [reply]
Sounds likely; thanks for sharing this @jmezach 👍
